h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -113,9 +131,10 @@ public void run() {
             List<RabitWorker> pending = new ArrayList<>();
             Queue<Integer> todoNodes = new ArrayDeque<>(tracker.workers);
             while (!interrupted() && shutdown.size() != tracker.workers) {
-                try {
-                    SocketChannel channel = tracker.sock.accept();
-                    RabitWorker worker = new RabitWorker(channel);
+                try{
+                    final SocketChannel channel = tracker.sock.accept(); // Does not proceed when interrupt() is called.
```

Comment:
Calling interrupt() on this thread will cause an exception to be thrown if the thread is waiting to accept new connections. However, this is caught by IOException catch block. 
The connection resources will be freed anyway, as a dedicated method is called in `RabitTracker` stop() method.